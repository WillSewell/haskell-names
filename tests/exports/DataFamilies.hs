{-# LANGUAGE TypeFamilies #-}
module DataFamilies (
    Vector,
    ListLike(..)) where

data family Vector a

class ListLike a where
  type I a
  h :: a -> I a

newtype instance Vector () = U_Vector ()
