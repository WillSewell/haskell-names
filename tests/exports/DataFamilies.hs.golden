[ DataFam
    { symbolModule = ModuleName "DataFamilies"
    , symbolName = Ident "Vector"
    , associate = Nothing
    }
, Class
    { symbolModule = ModuleName "DataFamilies"
    , symbolName = Ident "ListLike"
    }
, TypeFam
    { symbolModule = ModuleName "DataFamilies"
    , symbolName = Ident "I"
    , associate = Just (Ident "ListLike")
    }
, Method
    { symbolModule = ModuleName "DataFamilies"
    , symbolName = Ident "h"
    , className = Ident "ListLike"
    }
, Constructor
    { symbolModule = ModuleName "DataFamilies"
    , symbolName = Ident "U_Vector"
    , typeName = Ident "Vector"
    }
]